tests/test_speed.py

```diff
@@ -0,0 +1,66 @@
+from swsscommon import swsscommon
+import time
+import re
+import json
+import os
+
+class TestSpeedSet(object):
+    num_ports = 32
+    def test_SpeedAndBufferSet(self, dvs):
+        speed_list = ['50000', '25000', '40000', '10000', '100000']
+
+        cdb = swsscommon.DBConnector(4, dvs.redis_sock, 0)
+        adb = swsscommon.DBConnector(1, dvs.redis_sock, 0)
+        cfg_port_table = swsscommon.Table(cdb, "PORT", '|')
+        cfg_buffer_profile_table = swsscommon.Table(cdb, "BUFFER_PROFILE", '|')
+        cfg_buffer_pg_table = swsscommon.Table(cdb, "BUFFER_PG", '|')
+        asic_port_table = swsscommon.Table(adb, "ASIC_STATE:SAI_OBJECT_TYPE_PORT")
+        asic_profile_table = swsscommon.Table(adb, "ASIC_STATE:SAI_OBJECT_TYPE_BUFFER_PROFILE")
+
+        buffer_profiles = cfg_buffer_profile_table.getKeys()
+        expected_buffer_profiles_num = len(buffer_profiles)
+        assert expected_buffer_profiles_num == 7
```

Comment:
Using number 7 here seems to be a recipe for long investigation when test fails. How can we make it clearer where this number came from and when test failed here, what to look for?

A comment will be a good start.

Reply:
it comes from the buffers profile used for the test
I think it can be calculates (something like `cat ... | grep "PROFILE" | wc -l`)
I'll check and update this comment